File: backend/src/main/java/hu/ibc/ibcpartners/core/dto/AuditLogDto.java
package hu.ibc.ibcpartners.core.dto;

import com.fasterxml.jackson.databind.JsonNode;
import hu.ibc.ibcpartners.core.entity.AuditEventType;

import java.time.Instant;
import java.util.List;

public record AuditLogDto(
        Long id,
        AuditEventType eventType,
        String entityType,
        Long entityId,
        String description,
        JsonNode before,
        JsonNode after,
        List<AuditChangeDto> changes,
        Long createdBy,
        String createdByName,
        Instant createdAt) {
}

File: frontend/src/app/features/dashboard/audit-log/audit-log-dto.ts
import {AuditEventType} from '../../../shared/audit-event-type';
import {AuditChangeDto} from './audit-change-dto';

export interface AuditLogDto {
  id: number;
  eventType: AuditEventType;
  entityType: string;
  entityId: number;
  description: string;
  before: unknown;
  after: unknown;
  changes: AuditChangeDto[];
  createdBy: number;
  createdByName: string;
  createdAt: string;
}
